main/main.c

```diff
@@ -91,79 +99,86 @@ esp_err_t setup(spi_device_handle_t* rfid_spi_handle) {
 	PASS_ERROR(spi2_init(), "Could not initialize SPI2 Host");
 	PASS_ERROR(rfid_init(rfid_spi_handle), "Could not add RFID to SPI Host");
 
+	PASS_ERROR(wifi_init(&on_wifi_connect), "Unable to init WiFi");
+
 	return ESP_OK;
 }
 
-void app_main(void) {
-	int index = 0;
-
-	esp_err_t message = adc_init();
-	if (message != ESP_OK) {
-		while (1) {
-			LOG("ADC1 Init Error");
-		}
+/**
+ * A simple rewrite of memcpy_s.
+ *
+ * It copies over bytes from one byte array to another byte array.
+ *
+ * This function takes in two lengths, the maximum amount of bytes copied over
+ * is the minimum of both lengths.
+ *
+ * @param dest The destination pointer
+ * @param dest_count How many bytes it should write too
+ * @param src The source pointer
+ * @param src_count How many bytes it should copy from
+ */
+void byte_copy(
+	uint8_t* dest, uint8_t dest_count, uint8_t* src, uint8_t src_count
+) {
+	uint8_t max_copy = dest_count < src_count ? dest_count : src_count;
+	for (uint8_t index = 0; index < max_copy; index++) {
+		dest[index] = src[index];
 	}
+}
 
-	spi_device_handle_t rfid_handle = {0};
-	setup(&rfid_handle);
-
-	// Wake up the rfid reader
-	rfid_wakeup_mifare_tag(&rfid_handle);
-
+esp_err_t get_and_store_credentials(spi_device_handle_t* handle) {
+	PASS_ERROR(rfid_wakeup_mifare_tag(handle), "Unable to wake-up MiFare tag");
 	tag_data_t tag = ndef_create_type();
+	PASS_ERROR(ndef_full_scan(handle, &tag), "Unable to scan MiFare tag");
 
-	ndef_extract_all_records(&rfid_handle, &tag);
+	connection_data_t connection_data;
+	picc_get_ssid(&tag, connection_data.ssid);
```

Comment:
Was het niet beter om gewoon de functie picc_get_data() functie te maken, aangezien die functies nergens anders worden aangeroepen?

Reply:
If it only gets used once, that's more reason *not* to make it a function. Why make it a function if we only use it once? Saves a few keystrokes and less mind games to find the origin of the code.